migration/src/test/java/org/axonframework/migration/AxonJakartaTest.java

```diff
@@ -0,0 +1,80 @@
+package org.axonframework.migration;
+
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+import static org.openrewrite.maven.Assertions.pomXml;
+
+public class AxonJakartaTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec
+                .parser(JavaParser.fromJavaVersion()
+                        .logCompilationWarningsAndErrors(true)
+                        .classpath("axon-messaging", "rewrite-spring"))
+                .recipe("/META-INF/rewrite/axon-jakarta.yml", "org.axonframework.axon.UpgradeAxonFramework_4_7_Jakarta");
+    }
+
+    @Test
+    void migrateImports() {
+        //language=java
+        rewriteRun(
+                java(
+                        "package sample.axon;\n" +
+                                "import org.axonframework.eventhandling.deadletter.legacyjpa.DeadLetterJpaConverter;\n" +
+                                "class ATest {\n" +
+                                "   DeadLetterJpaConverter converter;\n" +
+                                "}",
+
+                        "package sample.axon;\n\n" +
+                                "import org.axonframework.eventhandling.deadletter.jpa.DeadLetterJpaConverter;\n\n" +
+                                "class ATest {\n" +
+                                "   DeadLetterJpaConverter converter;\n" +
+                                "}"
```

Comment:
These types of tests would look a whole lot better if for `src/test/java` we'd switch over to Java 17 & text blocks. Didn't know if that would go over well for this module, so figured keep it simple & ugly for now.

Reply:
I would prefer JDK17 (with text blocks) ASAP too. 
Granted, there's a ton of work to process before we're there for Axon Framework. 
So, easier said than done.

Regardless, thanks for the insights, of course.
I agree with the format _this_ is currently in.

Reply:
With some work it's possible to [use a different version of Java for the tests](https://vladmihalcea.com/different-java-main-test-maven/) only, but that might affect how you do CI for instance, so might not be an option still. (And you likely already knew that, but on the slim case that you didn't)

Reply:
I was aware, thanks! :-) But yes, I foresee impact for the CI/release process. So, to keep things simple (albeit ugly for these tests), I'd like to stick to JDK8 for tests.